src/khoj/processor/embeddings.py

```diff
@@ -28,13 +36,30 @@ def __init__(
 
     def embed_query(self, query):
         if self.api_key is not None and self.inference_endpoint is not None:
-            target_url = f"{self.inference_endpoint}"
-            payload = {"inputs": [query]}
-            headers = {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
-            response = requests.post(target_url, json=payload, headers=headers)
-            return response.json()["embeddings"][0]
+            return self.embed_with_api([query])[0]
         return self.embeddings_model.encode([query], show_progress_bar=False, **self.encode_kwargs)[0]
 
+    @retry(
+        retry=retry_if_exception_type(Exception),
```

Comment:
Do we want to limit retry to when there is an HTTPError? This would avoid retrying on non-recoverable errors (like a bug in our code, or a change in their API's response format)

```suggestion
        retry=retry_if_exception_type(requests.exceptions.HTTPError),
```